app/controllers/MessageController.scala

```diff
@@ -37,6 +39,19 @@ class MessageController @Inject()(
     (messageRecipientIdentifier() andThen messageTypeIdentifier()).async(parse.xml) { implicit request =>
       routingService
         .sendMessage(request.messageRecipient, request.directable, request.body, request.headers)
-        .map(response => Status(response.status))
+        .map {
+          response =>
+            response.status match {
+              case status if is2xx(status) =>
+                response.header(LOCATION) match {
+                  case Some(value) =>
+                    Status(response.status).withHeaders(LOCATION -> value)
+                  case None =>
+                    InternalServerError
```

Comment:
I don't believe that this is the right status to return here, I believe that this should be Ok response with no header value for location